File: Ruby/demos/browser/features/support/pages/homepage.rb
module Pages
  def homepage
    @homepage ||= Homepage.new
  end

  class Homepage < TestEvolve::Core::PageObject
    element(:my_banner) { div(class: 'sqs-cookie-banner-v2-acceptWrapper') }
    element(:my_link) { a(data_test: 'example-link') }
    element(:my_button) { button(data_test: 'example-button') }
    element(:text) { text_field(data_test: 'example-text') }
    element(:phone) { text_field(data_test: 'example-phone') }
    element(:date_time) { date_time_field(data_test: 'example-datetime') }
    element(:date) { date_field(data_test: 'example-date') }
    element(:week) { input(data_test: 'example-week') }
    element(:month) { input(data_test: 'example-month') }
    element(:range) { input(data_test: 'example-range') }
    element(:file) { file_field(data_test: 'example-file') }
    element(:url) { text_field(name: 'url') }
    element(:my_search) { text_field(data_test: 'example-search') }
    element(:colour) { input(data_test: 'example-colour1') }
    element(:password) { text_field(data_test: 'example-password') }
    element(:number) { text_field(data_test: 'example-number') }
    element(:my_radio) { radio(data_test: 'example-radio-1') }
    element(:my_radio_2) { radio(data_test: 'example-radio-2') }
    element(:my_checkbox) { checkbox(data_test: 'example-checkbox') }
    element(:my_select) { select(data_test: 'example-select') }
    element(:button_double) { button(data_test: 'example-double') }
    element(:email) { text_field(data_test: 'example-email') }

    def interact_with_page_objects()
      my_banner.click
      my_link.click
      my_button.click
      text.set('some text')
      raise "Expected: 'some text', Actual: #{text.value}" unless text.value == 'some text'
      phone.set('01268 467890')
      date_time.set('2025-02-02T13:57')
      date.set('2025-02-02')
      week.set('262025')
      month.set('11')
      month.send_keys :tab
      month.set('2025')
      2.times do
        range.send_keys :right
      end
      range.send_keys :left
      file.set("#{Dir.pwd}/features/support/resources/robot_picture.png")
      url.set('www.bbc.co.uk')
      my_search.set('looking for...')
      raise "Expected: 'looking for...', Actual: #{my_search.value}" unless my_search.value == 'looking for...'
      colour.set('#00ff41')
      password.set('my secret')
      raise "Expected: 'my secret', Actual: #{password.value}" unless password.value == 'my secret'
      number.set("12345")
      my_radio.scroll.to
      my_radio.click
      my_radio_2.click
      my_checkbox.check
      my_select.select('Option 3')
      button_double.double_click
      email.set('test@test.com')
    end

    def assert_object_changes
      raise 'Failure - Element still displayed' if my_banner.present?
    end
  end
end

File: Python/demos/browser/features/support/pages/homepage.py
from testevolve import TestEvolve
from testevolve.browser.page_object import PageObject

import os

class Homepage(PageObject):
    def __init__(self):
        self.accept_cookies_button = self.element(class_name='accept')
        self.myLink = self.element(name='example-link')
        self.myButton = self.element(name='example-button')
        self.myText = self.element(name='example-text')
        self.phone = self.element(name='example-phone')
        self.dateTime = self.element(name='example-datetime')
        self.date = self.element(name="example-date")
        self.week = self.element(name="example-week")
        self.month = self.element(name="example-month")
        self.range = self.element(name="example-range")
        self.file = self.element(name="example-file")
        self.url = self.element(name="url")
        self.mySearch = self.element(name="example-search")
        self.colour = self.element(name="example-colour1")
        self.password = self.element(name="example-password")
        self.number = self.element(name="example-number")
        self.myRadio = self.element(name="example-radio-1")
        self.myCheckbox = self.element(name="example-checkbox")
        self.mySelect = self.element(name="example-select")
        self.buttonDouble = self.element(name="example-double")
        self.email = self.element(name="example-email")

    def open(self):
        TestEvolve.browser.get("https://testevolve.com/element-capture")

    def open_page(page):
        TestEvolve.browser.get(page)

    def accept_cookies(self):
        try:
            self.accept_cookes_button().is_displayed()
            self.accept_cookes_button().click()
        except:
            TestEvolve.log.debug('Accept cookies button not found. Continuing')

    def interact_with_page_objects(self):
        self.myLink().click()
        self.myButton().click()
        self.myText().send_keys('some text')
        self.phone().send_keys('01268 467890')
        self.dateTime().send_keys('2025-02-02', '13:57')
        self.date().send_keys('2025-02-02')
        self.week().send_keys('26','2025')
        self.month().send_keys('12', '2025')
        self.range().send_keys('3')
        self.range().send_keys('5')
        self.range().send_keys('4')
        self.file().send_keys(os.getcwd()+"/features/support/resources/robot_picture.png")
        self.url().send_keys('www.bbc.co.uk')
        self.mySearch().send_keys('looking for...')
        self.colour().send_keys('#00ff41')
        self.password().send_keys('my secret')
        self.number().send_keys("12345")
        TestEvolve.browser.scroll_into_view(self.myRadio())
        self.myRadio().click()
        TestEvolve.browser.scroll_into_view(self.myCheckbox())
        self.myCheckbox().click()
        self.mySelect.select_value('Option 3')
        self.buttonDouble.double_click()
        self.email().send_keys('test@test.com')
